spring-cloud-aws-dynamodb/src/main/java/io/awspring/cloud/dynamodb/DefaultDynamoDbTableNameResolver.java

```diff
@@ -16,19 +16,32 @@
 package io.awspring.cloud.dynamodb;
 
 import java.util.Locale;
+import org.springframework.core.env.Environment;
 import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
 
 /**
  * Simple implementation of {@link DynamoDbTableNameResolver} that resolves class simple name to table name.
  *
  * @author Matej Nedic
+ * @author Arun Patra
  * @since 3.0
  */
 public class DefaultDynamoDbTableNameResolver implements DynamoDbTableNameResolver {
 
+	private final Environment environment;
+
+	public DefaultDynamoDbTableNameResolver(Environment environment) {
```

Comment:
Shouldn't the `DynamoDbProperties` be injected here rather than `Environment`?

Reply:
@maciejwalkowiak The `DynamoDbProperties` are in the `spring-cloud-aws-autoconfigure` module, and the dependency graph is pointing from `spring-cloud-aws-dynamodb` towards that. 
Actually if memory serves right, some time back we had been talking on the dependency graph where the motivation was.

1. Keep the lower level modules fairly de-coupled from Spring Boot per se, so that thay can be used by users who do not use Spring Boot and/or do not want to use auto-configuration.
2. We ended up following a uniform patterm where the properties are in the `spring-cloud-aws-autoconfigure` module with the intent that the properties are primarily meant to manage/customize auto-configuration.

So, I had to infer the properties via the `Environment`,

Reply:
But `DynamoDefaultDbTableNameResolver` and `DynamoDbProperties` are in different modules and it would mean creating a cyclic dependency if we try injecting the properties. The only option left would be to pull out `DynamoDbProperties` from `spring-cloud-aws-autoconfigure` and move  it to `spring-cloud-aws-dynamodb`; this would be a departurture from the other modules. Secondly, relocating the `DynamoDbProperties` would be a breaking build time change for users of the RC release when we go GA.

Another thing is, `DynamoDefaultDbTableNameResolver` is a core construct necessary for the `DynamoDbTemplate` to work (whether you auto-configure or not). Hence, `DynamoDefaultDbTableNameResolver` has to be in the lower level module `spring-cloud-aws-dynamodb`.

Honestly I did not want to inject the `Environment` at all given the fact it sounds a bit off to do so, especially given the fact what we are using it for. But is there any other way in this case?

Reply:
i mean to pass prefix as a string and use dynamodbproperties in the autoconfigure module

Reply:
Aaah, I see what you mean. Done. I refactored `DynamoDefaultDbTableNameResolver` so that it does not need an `Environment` injection, instead it takes a `tablePrefix`. 

There's one special case though, which we will have to accomodate. A user wanting a  _raw_ `DynamoDbTemplate` by supplying their own `DynamoDbEnhancedClient` needs a `DefaultDynamoDbTableNameResolver` too. In that case, we would need to inject the `Environment` into the `DynamoDbTemplate` if we were to offer the benefits of the `tablePrefix`. But I guess, that should be acceptable. If we don't do this, then we will have to give the user a plain and un-decorated `DefaultDynamoDbTableNameResolver` which is not aware of the `tablePrefix`. This is just a fallout of our design and dependency graph, nothing else. IMO, this should be ok. What do you think?